File: samples/InfoCardRelyingParty/App_Code/TracePageAppender.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Web;

public class TracePageAppender : log4net.Appender.AppenderSkeleton {
	protected override void Append(log4net.Core.LoggingEvent loggingEvent) {
		StringWriter sw = new StringWriter(Logging.LogMessages);
		Layout.Format(sw, loggingEvent);
	}
}

File: samples/OpenIdRelyingPartyWebFormsVB/Code/TracePageAppender.vb
Imports System.IO

Public Class TracePageAppender
	Inherits log4net.Appender.AppenderSkeleton

	Protected Overrides Sub Append(ByVal loggingEvent As log4net.Core.LoggingEvent)
		Dim sw As StringWriter = New StringWriter(Global_asax.LogMessages)
		Layout.Format(sw, loggingEvent)
	End Sub
End Class